middle_end/flambda2/simplify_shared/closure_offsets.ml

```diff
@@ -16,18 +16,19 @@
 
 open! Flambda.Import
 
-let filter_closure_vars set ~used_closure_vars =
-  let closure_elements = Set_of_closures.closure_elements set in
+let keep_closure_var ~used_closure_vars v =
   match (used_closure_vars : _ Or_unknown.t) with
-  | Unknown -> closure_elements
+  | Unknown -> true
   | Known used_closure_vars ->
-    Var_within_closure.Map.filter
-      (fun clos_var _bound_to ->
-        if Compilation_unit.is_current
-             (Var_within_closure.get_compilation_unit clos_var)
-        then Var_within_closure.Set.mem clos_var used_closure_vars
-        else true)
-      closure_elements
+    if Compilation_unit.is_current (Var_within_closure.get_compilation_unit v)
+    then Var_within_closure.Set.mem v used_closure_vars
+    else true
+
+let filter_closure_vars set ~used_closure_vars =
+  let closure_elements = Set_of_closures.closure_elements set in
+  Var_within_closure.Map.filter
+    (fun v _ -> keep_closure_var ~used_closure_vars v)
+    closure_elements
```

Comment:
Isn't the new implementation of `filter_closure_vars` the same as the old one, but less efficient ? You could keep the same structure if you want but the match on `used_closure_vars` deserves to be lifted out of the map.

Reply:
You're right, I'll fix that.

Reply:
done in 2ff2cf7